packages/web-components/src/components/va-combo-box/va-combo-box.tsx

```diff
@@ -0,0 +1,251 @@
+  State,
+  Fragment,
+} from '@stencil/core';
+import classnames from 'classnames';
+import { i18next } from '../..';
+import { comboBox } from './va-combo-box-library.js';
+
+/**
+ * @componentName Combo Box
+ * @maturityCategory dont_use
+ * @maturityLevel proposed
+ * @guidanceHref form/combo-box
+ * @translations English
+ * @translations Spanish
+ */
+
+@Component({
+  tag: 'va-combo-box',
+  styleUrl: 'va-combo-box.scss',
+  shadow: true,
+})
+export class VaComboBox {
+  @Element() el: any;
+
+  @State() options: Array<Node>;
+
+  @State() labelNode: HTMLLabelElement;
+
+  /**
+   * Whether or not this is a required field.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * The combo box component will be disabled / read-only.
+   */
+  @Prop() disabled?: boolean = false;
+
+  /**
+   * Text label for the field.
+   */
+  @Prop() label!: string;
+
+  /**
+   * The placeholder string.
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * Name attribute for the select field.
+   */
+  @Prop() name!: string;
+
+  /**
+   * Selected value (will get updated on select).
+   */
+  @Prop({ reflect: true, mutable: true }) value?: string;
+
+  /**
+   * Error message to display. When defined, this indicates an error.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Whether or not `aria-invalid` will be set on the inner select. Useful when
+   * composing the component into something larger, like a date component.
+   */
+  @Prop() invalid?: boolean = false;
+
+  /**
+   * Whether or not to fire the analytics events
+   */
+  @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * Optional hint text.
+   */
+  @Prop() hint?: string;
+
+  /**
+   * An optional message that will be read by screen readers when the select is focused.
+   */
+  @Prop() messageAriaDescribedby?: string;
+
+  /**
+   * The event emitted when the selected value changes
+   */
+  @Event() vaSelect: EventEmitter;
+
+  /**
+   * Whether an error message should be shown - set to false when this component is used inside va-date or va-memorable-date
+   * in which the error for the va-select will be rendered outside of va-select
+   */
+  @Prop() showError?: boolean = true;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * option is selected and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+    if (!this.label) {
+      throw new Error('va-combo-box: label is a required property');
+    }
+  }
+
+  componentDidLoad() {
+    const comboBoxElement = this.el.shadowRoot.querySelector('.usa-combo-box');
+    const labelElement = this.el.shadowRoot.querySelector('label');
+    if (comboBoxElement) {
+      comboBox.init(comboBoxElement, labelElement);
+      comboBox.on(comboBoxElement);
```

Comment:
I'm not sure I understand what the label needs to be added in this way instead of just in the markup. Do you mind clarifying?

Reply:
so this is where the tricky part with uswds combobox component comes in. I am actually not adding a label here, but rather passing a reference to the `init` function. This is a slight change I hade to make to `init` function. Prior to this, the function was trying to identify a label with `document.querySelector` but it was not able to due to `label` in our case being rendered in `shadow-root`. This is just a workaround. `<label>` is actually added within `<Host>`

Reply:
@oleksii-morgun ah! my bad for not digging far enough into the `va-combo-box-library` script. Most of that script is directly from the USWDS? 

- https://github.com/uswds/uswds/blob/develop/packages/usa-combo-box/src/index.js

Reply:
correct, I just made couple minor changes to account for that label reference in `init` function. the rest is copy-paste